core/services/pg/locked_db.go

```diff
@@ -139,10 +140,46 @@ func (l lockedDb) DB() *sqlx.DB {
 	return l.db
 }
 
-func openDB(appID uuid.UUID, cfg LockedDBConfig) (db *sqlx.DB, err error) {
+func openDB(appID uuid.UUID, cfg LockedDBConfig, opts ...ConnectionOpt) (db *sqlx.DB, err error) {
 	uri := cfg.URL()
 	static.SetConsumerName(&uri, "App", &appID)
 	dialect := cfg.Dialect()
-	db, err = NewConnection(uri.String(), dialect, cfg)
+	connStr := uri.String()
+	for _, opt := range opts {
+		opt(&connStr)
+	}
+	db, err = NewConnection(connStr, dialect, cfg)
 	return
 }
+
+type ConnectionOpt func(*string)
```

Comment:
This shouldn't just operate on the uri IMO, it should operate on something higher level, like a LockedDBConfig that's been turned into a struct.